.gitignore

```diff
@@ -133,3 +133,6 @@ arc-mlir/.guessed-arc-mlir
 
 # Editor backup files
 *~
+build/
+CMakeCache.txt
+CMakeFiles/
```

Comment:
The only reason for `CMakeCache.txt` and `CMakeFiles/` to occur in your source directory is if you did something wrong, and that could break proper builds in `build/`. So I think it is better to only add `build/`, that way we'll notice that something is wrong.